cmd/skaffold/app/cmd/filter.go

```diff
@@ -64,7 +65,12 @@ func runFilter(ctx context.Context, out io.Writer, debuggingFilters bool, buildA
 		if err != nil {
 			return err
 		}
-		manifestList := manifest.ManifestList([][]byte{bytes})
+
+		var manifestList manifest.ManifestList
+		for _, manifestStr := range strings.Split(string(bytes), "---") {
```

Comment:
This will catch `---` where it occurs within a string.  It needs to be preceded and followed by newlines.

Reply:
Good catch